source/dofs/dof_handler_policy.cc

```diff
@@ -4359,18 +4362,80 @@ namespace internal
       template <class DoFHandlerType>
       NumberCache
       ParallelDistributed<DoFHandlerType>::
-      renumber_mg_dofs (const unsigned int                          /*level*/,
-                        const std::vector<types::global_dof_index> &/*new_numbers*/) const
+      renumber_mg_dofs (const unsigned int                          level,
+                        const std::vector<types::global_dof_index> &new_numbers) const
       {
-        // this is not currently implemented, but should be simple to do by
-        // just calling the function like in the sequential case just with
-        // an appropriate index set argument
-        Assert(false, ExcNotImplemented());
+        // we only implement the case where the multigrid numbers are
+        // renumbered within the processor's partition, rather than the most
+        // general case
+        const std::vector<IndexSet> index_sets = dof_handler->locally_owned_mg_dofs_per_processor(level);
+
+        constexpr int dim = DoFHandlerType::dimension;
+        constexpr int spacedim = DoFHandlerType::space_dimension;
+        const parallel::Triangulation<dim, spacedim> *tr =
+          (dynamic_cast<const parallel::Triangulation<dim, spacedim>*>
+           (&this->dof_handler->get_triangulation()));
+        Assert(tr != nullptr, ExcInternalError());
+        const unsigned int my_rank = Utilities::MPI::this_mpi_process(MPI_COMM_WORLD);
 
-        return NumberCache ();
-      }
+        for (types::global_dof_index i : new_numbers)
+          {
+            (void)i;
+            Assert(index_sets[my_rank].is_element(i),
+                   ExcNotImplemented("Renumberings that change the locally owned mg dofs "
+                                     "partitioning are currently not implemented for "
+                                     "the multigrid levels"));
+          }
 
+        // we need to access all locally relevant degrees of freedom. we
+        // use Utilities::MPI::Partitioner for handling the data exchange
+        // of the new numbers, which is simply the extraction of ghost data
+        IndexSet relevant_dofs;
+        DoFTools::extract_locally_relevant_level_dofs(*dof_handler, level, relevant_dofs);
+        std::vector<types::global_dof_index> ghosted_new_numbers(relevant_dofs.n_elements());
+        {
+          Utilities::MPI::Partitioner partitioner(index_sets[my_rank],
+                                                  relevant_dofs, tr->get_communicator());
+          std::vector<types::global_dof_index> temp_array(partitioner.n_import_indices());
+          std::vector<MPI_Request> requests;
+          partitioner.export_to_ghosted_array_start
+          (17, make_array_view(new_numbers), make_array_view(temp_array),
```

Comment:
just to ease the reading, maybe do `const unsigned int communication_channel=17`.